### DISCUSSION TITLE
Any ideas how to work around composer auto-scripts always failing when .env references Docker secret file paths?

### DISCUSSION THREAD

**Discussion by jerrac** [symfony/symfony#48973] [Q&A] [answered]
When you set up a new project based on the Getting Started docs, this is put in your composer.json:

```json
"scripts": {
        "auto-scripts": {
            "cache:clear": "symfony-cmd",
            "assets:install %PUBLIC_DIR%": "symfony-cmd"
        },
        "post-install-cmd": [
            "@auto-scripts"
        ],
        "post-update-cmd": [
            "@auto-scripts"
        ]
    },
```
If you happen to use Docker secrets, like:

.env:
```
DB_NAME="/run/secrets/db_name"
```
config/packages/doctrine.yaml
```yaml
# ...
connections:
      default:
        driver: 'pdo_mysql'
        server_version: 'mariadb-10.9.4'
        dbname: '%env(file:DB_NAME)%'
# ...
```
composer will throw errors like:
```
Executing script assets:install public [KO]
 [KO]
Script assets:install %PUBLIC_DIR% returned with error code 1
!!  
!!  In EnvVarProcessor.php line 136:
!!                                                                                 
!!    File "/run/secrets/db_name" not found (resolved from "DB_NAME"  
!!    ).                                                                           
!!                                                                                 
!!  
!!  
Script @auto-scripts was called via post-update-cmd
```
every time you run an update/install/require command. Often it will then remove the package you just added to your composer.json because it thinks the install failed.

I can do a partial work around by putting composer.phar in my Symfony bin dir, and running it from inside the app container. 

But that isn't going to work when I set up my CI workflow. I generally run composer outside of Docker to install dependencies, and then build the image. 

Anyone have any other ideas?

**Comment by MatTheCat** [comment]
You really should run composer in the same environment than your project, as any difference will potentially trigger issues.

**Comment by jerrac** [comment, marked as answer]
Thanks to some help on Slack, I added `-e APP_ENV=prod` to my docker command it now works.
